L1Trigger/L1THGCal/src/backend/HGCalShowerShape.cc

```diff
@@ -557,25 +560,34 @@ int HGCalShowerShape::bitmap(const l1t::HGCalMulticluster& c3d, int start, int e
       if (!pass(*id_tc.second, c3d))
         continue;
       unsigned layer = triggerTools_.triggerLayer(id_tc.second->detId());
-      if (layer ==0 || layer > nlayers) continue;
-      layers[layer - 1] += id_tc.second->pt(); //shift by -1 because layer 0 doesn't exist
+      if (layer == 0 || layer > nlayers)
+        continue;
+      layers[layer - 1] += id_tc.second->pt();  //shift by -1 because layer 0 doesn't exist
     }
   }
   double tot = 0;
   for (unsigned i = 0; i < layers.size(); ++i)
     tot += layers[i];
   uint32_t bitmap = 0;
-  for (int i = start-1; i <= end-1; i++) {
-    if ((end-i) >32) continue;
-    bitmap += (layers[i] >= threshold)<<(end-(i+1));
+  int bitmap_size = 32;
+  if (start == 0) {
+    edm::LogWarning("Layer0") << "layer 0 doesn't exist, defaulted start to layer 1";
```

Comment:
The Warning message category (the type of warning) is not very informative.
Please have a look here:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger#Guidelines_for_choosing_category
regarding the guidelines to choose a message category